subxt/src/tx/tx_client.rs

```diff
@@ -465,4 +467,35 @@ where
         let dry_run_bytes = self.client.rpc().dry_run(self.encoded(), at).await?;
         dry_run_bytes.into_dry_run_result(&self.client.metadata())
     }
+
+    /// This returns an estimate for what the extrinsic is expected to cost to execute, less any tips.
+    /// The actual amount paid can vary from block to block based on node traffic and other factors.
+    pub async fn partial_fee_estimate(&self) -> Result<u128, Error> {
+        /// adapted from original type <https://paritytech.github.io/substrate/master/pallet_transaction_payment_rpc_runtime_api/struct.RuntimeDispatchInfo.html>
+        #[derive(Debug, Encode, Decode, Eq, PartialEq, Copy, Clone)]
+        pub struct RuntimeDispatchInfo {
+            /// Weight of this dispatch. Adapted from original type <https://paritytech.github.io/substrate/master/frame_support/weights/struct.Weight.html>
+            #[codec(compact)]
+            weight_ref_time: u64,
+            #[codec(compact)]
+            weight_proof_size: u64,
+            /// Class of this dispatch; an enum variant index. Adapted from original type <https://paritytech.github.io/substrate/master/frame_support/dispatch/enum.DispatchClass.html>
+            class: u8,
+            /// The inclusion fee of this dispatch; what we want:
+            partial_fee: u128,
```

Comment:
I think the computing cost is so marginal, that I think the complexity increase and understandability descrease in doing so, might not be worth it.

Reply:
Personally, I don't like these extra fields which makes it a bit harder to understand and reason about.

I did [something similar in the staking-miner](https://github.com/paritytech/staking-miner-v2/blob/main/src/epm.rs#L273-#L290) where `RuntimeDispatchInfo` only has the weight field but very much like this

Reply:
It would be good to see if the contract team uses the weight estimation as well. In that case, it might be worth adding it since we'd have more feedback and data points around it

Reply:
Ok, I see then.

Thus, we should mimic the type exactly as in substrate with same fields and so on or just mark the fields that we don't care about such as:

```rust    
        /// adapted from original type <https://paritytech.github.io/substrate/master/pallet_transaction_payment_rpc_runtime_api/struct.RuntimeDispatchInfo.html>   
        #[derive(Debug, Encode, Decode, Eq, PartialEq, Copy, Clone)]
        struct RuntimeDispatchInfo {
             /// This field is just used to make Compact encoding work.
             _: Weight,
              /// This field is just used to make Compact encoding work.
             _: DispatchClass,
             partial_fee: u128,
        }
```

Reply:
I'm easy either way, but leant towards removing the "noise" of 2 additional struct/enum definitions :)

When the Runtime API stuff is in metadata for real we can improve on all of this anyway